### PR TITLE
 [otbn, dv] otbn_imem_err test added along with related changes

### LINKED ISSUE #8505: [otbn/dv] Corrupt IMEM in testing
Corrupt IMEM in testing and ensure that we see a fatal error.

> estimate 16
> remaining 2021-10-18 2

### PR DESCRIPTION
**Pull request by prajwalaputtappa**
This is in draft because it depends on https://github.com/lowRISC/opentitan/pull/8600 and https://github.com/lowRISC/opentitan/pull/8401. 
Once the above two PRs are merged, only the final commit will remain. 

The following changes have been made in this commit:
1. otbn_imem_err_vseq.sv: This is the sequence required to run otbn_imem_err
test. This is a new file.
2. otbn_scoreboard.sv: otbn_scoreboard is taught to expect a fatal error
when the model status goes to StatusLocked.
3. otbn_base_vseq.sv: run_otbn function is made to stall until otbn status is not
StatusBusyExecute instead of waiting for otbn to go to idle.
4. otbn_model_if.sv: wait_status now waits on negedge of the clock
instead of posedge to avoid race condition between alert arrival and
model state change. It ensures that the scoreboard sees the state change
before the fatal alert changes and knows it has to expect a fatal alert
and will not throw a fatal error.

Fixes #8505

### CODE REVIEW

File: hw/ip/otbn/dv/uvm/env/seq_lib/otbn_imem_err_vseq.sv
Comment (rswarbrick) [inline hw/ip/otbn/dv/uvm/env/seq_lib/otbn_imem_err_vseq.sv:30]: Would you mind renaming this to something like `wait_cycles`? The name comes from the reset sequence, where it's the amount of time we wait before resetting the block. It's probably not quite the right name here.

File: hw/ip/otbn/dv/uvm/env/otbn_scoreboard.sv
Comment (rswarbrick) [inline hw/ip/otbn/dv/uvm/env/otbn_scoreboard.sv:284]: Super-nit: I'd consider dropping these two extra lines if you're re-pushing again anyway.

**Review by rswarbrick — APPROVED** [review]
This looks good! Obviously, we have to wait until the PRs it depends on have merged but, once they are in, I think this should be good to go.

File: hw/ip/otbn/dv/uvm/env/otbn_scoreboard.sv
Comment (prajwalaputtappa) [reply hw/ip/otbn/dv/uvm/env/otbn_scoreboard.sv]: Done

**Comment by imphil** [comment]
All dependencies have gone in now, @prajwalaputtappa can you rebase this PR please?

File: hw/ip/otbn/dv/uvm/env/seq_lib/otbn_imem_err_vseq.sv
Comment (rswarbrick) [reply hw/ip/otbn/dv/uvm/env/seq_lib/otbn_imem_err_vseq.sv:30]: I think this is still called `reset_wait`: did something get dropped in a rebase?

File: hw/ip/otbn/dv/uvm/env/seq_lib/otbn_imem_err_vseq.sv
Comment (rswarbrick) [inline hw/ip/otbn/dv/uvm/env/seq_lib/otbn_imem_err_vseq.sv:33]: Nit "1 and" (missing space)

File: hw/ip/otbn/dv/uvm/env/seq_lib/otbn_imem_err_vseq.sv
Comment (rswarbrick) [inline hw/ip/otbn/dv/uvm/env/seq_lib/otbn_imem_err_vseq.sv:50]: I know this text came from me, but the grammar's a bit wonky! Could you change it to "we know that either..." ?

File: hw/ip/otbn/dv/uvm/env/seq_lib/otbn_imem_err_vseq.sv
Comment (rswarbrick) [inline hw/ip/otbn/dv/uvm/env/seq_lib/otbn_imem_err_vseq.sv:104]: Nit: Missing space before `.value`

Comment (prajwalaputtappa) [reply hw/ip/otbn/dv/uvm/env/seq_lib/otbn_imem_err_vseq.sv]: Made all the changes and pushed it again

**Review by rswarbrick — APPROVED** [review]
Looks great!

**Pull request merged**
